src/logic/schedule-exporter/workers-absence-export.logic.ts

```diff
@@ -0,0 +1,199 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import xlsx from "exceljs";
+import { ScheduleDataModel } from "../../state/schedule-data/schedule-data.model";
+import { EMPTY_ROW, ABSENCE_HEADERS } from "../../helpers/parser.helper";
+import { CELL_MARGIN } from "./schedule-export.logic";
+import { ShiftCode, SHIFTS } from "../../state/schedule-data/shifts-types/shift-types.model";
+import { TranslationHelper } from "../../helpers/translations.helper";
+import {
+  DateInformationForWorkInfoCalculation,
+  WorkerHourInfo,
+} from "../schedule-logic/worker-hours-info.logic";
+import { PrimaryMonthRevisionDataModel } from "../../state/application-state.model";
+import { MonthInfoLogic } from "../schedule-logic/month-info.logic";
+import {
+  VerboseDate,
+  WeekDay,
+} from "../../state/schedule-data/foundation-info/foundation-info.model";
+import { ContractType } from "../../state/schedule-data/worker-info/worker-info.model";
+
+interface AbsenceInfo {
+  workersAbsenceInfoArray: (string | number)[][];
+  cellsToMerge: number[][];
+}
+
+export class WorkersAbsenceExportLogic {
+  private scheduleModel: ScheduleDataModel;
+
+  private revision: PrimaryMonthRevisionDataModel;
+
+  constructor(scheduleModel: ScheduleDataModel, revision: PrimaryMonthRevisionDataModel) {
+    this.scheduleModel = scheduleModel;
+    this.revision = revision;
+  }
+
+  public setAbsenceWorkSheet(workSheet: xlsx.Worksheet): void {
+    workSheet.pageSetup.showGridLines = true;
+    workSheet.pageSetup.fitToPage = true;
+    workSheet.pageSetup.fitToHeight = 1;
+    workSheet.pageSetup.fitToWidth = 1;
+    workSheet.pageSetup.horizontalCentered = true;
+
+    const { workersAbsenceInfoArray, cellsToMerge } = this.createAbsenceInfoSection(
+      this.scheduleModel
+    );
+
+    const colLens = workersAbsenceInfoArray[0].map((_, colIndex) =>
+      Math.max(...workersAbsenceInfoArray.map((row) => row[colIndex]?.toString().length ?? 0))
+    );
+
+    workSheet.addRows(workersAbsenceInfoArray);
+
+    cellsToMerge.forEach((pair, index) => {
```

Comment:
Co prawda wcześniej o tym nie wspomniałem, ale ładnie to wyszło w tej drugiej funkcji, więc może i tę udałoby się tak podzielić na prywatne funkcje. Dużo łatwiej się to czyta.